packages/styled-system/src/create-theme-vars/create-theme-vars.ts

```diff
@@ -71,6 +76,96 @@ const tokenHandlerMap: Partial<Record<ThemeScale, TokenHandler>> & {
       },
     }
   },
+  tokensMap: (keys, value, options) => {
+    const [, scale, ...tokenPath] = keys
+    const latestKey = tokenPath[tokenPath.length - 1]
+    const isConditionToken = Object.keys(options.conditions ?? {}).includes(
+      latestKey,
+    )
+
+    const varKey = [scale, ...tokenPath].join("-")
+    const lookupKey = [scale, ...tokenPath].join(".")
+
+    if (!isConditionToken) {
+      const { variable, reference } = cssVar(
+        varKey,
+        undefined,
+        options.cssVarPrefix,
+      )
+
+      const cssVars = {
+        [variable]: cssVar(
+          [scale, ...String(value).split(".")].join("-"),
+          undefined,
+          options.cssVarPrefix,
+        ).reference,
+      }
+
+      return {
+        cssVars,
+        cssMap: {
+          [lookupKey]: {
+            value,
+            variable,
+            reference,
+          },
+        },
+      }
+    }
+
+    /**
+     * @example ["colors", "text"]
+     */
+    const parentPath = [scale, ...tokenPath.slice(0, -1)]
+    const parentVarKey = parentPath.join("-")
+    const parentLookupKey = parentPath.join(".")
+
+    const { variable, reference } = cssVar(
+      parentVarKey,
+      undefined,
+      options.cssVarPrefix,
+    )
+
+    /**
+     * Resolves `_dark` to `[data-theme=dark] &`
+     */
+    const resolvedCondition = String(
+      options.conditions?.[latestKey] ?? latestKey,
+    )
+
+    /**
+     * Resolves for `scale="colors", value="gray.100"` to `--colors-gray-100`
+     */
+    const targetCssVar = cssVar(
+      [scale, ...String(value).split(".")].join("-"),
+      undefined,
+      options.cssVarPrefix,
+    )
+
+    const cssVarValue = cssVar(
+      [...parentPath, "DEFAULT"].join("-"),
+      "var(--chakra-empty,/*!*/ /*!*/)",
```

Comment:
Can we create this css var using `cssVar` function and cleanup existing references?